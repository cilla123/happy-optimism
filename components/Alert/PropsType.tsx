export default interface PropsType {
  shape?: 'radius';
  visible?: boolean;
  animationType?:
    'fade' | 'door' | 'flip' | 'rotate' | 'zoom' |
    'moveUp' | 'moveDown' | 'moveLeft' | 'moveRight' |
    'slideUp' | 'slideDown' | 'slideLeft' | 'slideRight';
  animationDuration?: number;
  width?: string | number;
  onMaskClick?: () => void;
  title?: any;
  message?: any;
  cancelText?: any;
  onCancel?: () => void;
}

declare global {
  interface Window {
    zarmAlert?: any;
  }
}
