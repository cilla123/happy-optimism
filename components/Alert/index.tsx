import React, { PureComponent } from 'react';
// import ReactDOM from 'react-dom';
import classnames from 'classnames';
import PropsType from './PropsType';
import Modal from '../Modal';

export interface AlertProps extends PropsType {
  prefixCls?: string;
  className?: string;
}

export default class Alert extends PureComponent<AlertProps, {}> {

  static defaultProps = {
    prefixCls: 'za-alert',
    animationType: 'zoom',
    message: '',
    cancelText: '关闭',
  };

  // static show = (props) => {
  //   ReactDOM.render(<Alert {...props} visible />, window.zarmAlert);
  // }

  // static hide = () => {
  //   ReactDOM.render(<Alert visible={false} />, window.zarmAlert);
  // }

  render() {
    const { prefixCls, className, title, message, cancelText, onCancel, ...others } = this.props;
    const cls = classnames(prefixCls, className);

    return (
      <Modal className={cls} {...others}>
        <Modal.Header title={title} />
        <Modal.Body className={`${prefixCls}-body`}>
          {message}
        </Modal.Body>
        <Modal.Footer className={`${prefixCls}-footer`}>
          <a className={`${prefixCls}-footer-button`} onClick={onCancel}>{cancelText}</a>
        </Modal.Footer>
      </Modal>
    );
  }
}

// if (!window.zarmAlert) {
//   window.zarmAlert = document.createElement('div');
//   document.body.appendChild(window.zarmAlert);
// }

// ReactDOM.render(<Alert visible={false} />, window.zarmAlert);
