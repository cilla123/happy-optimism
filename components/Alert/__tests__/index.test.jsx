import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Alert from '../index';

describe('Alert', () => {
  it('renders correctly', () => {
    const onCancel = jest.fn();

    const wrapper = render(
      <Alert
        radius
        visible
        title="警告"
        message="这里是警告信息"
        onCancel={onCancel}
        />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
