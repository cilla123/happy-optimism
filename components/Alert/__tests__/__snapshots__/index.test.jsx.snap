// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders correctly 1`] = `
<div
  class="za-modal za-alert"
  style="-webkit-animation-duration:200ms;animation-duration:200ms"
>
  <div
    class="za-modal-wrapper"
  >
    <div
      class="za-modal-dialog"
      style="width:70%;-webkit-animation-duration:200ms;animation-duration:200ms"
    >
      <div
        class="za-modal-header"
      >
        <div
          class="za-modal-header-title"
        >
          警告
        </div>
      </div>
      <div
        class="za-modal-body za-alert-body"
      >
        这里是警告信息
      </div>
      <div
        class="za-modal-footer za-alert-footer"
      >
        <a
          class="za-alert-footer-button"
        >
          关闭
        </a>
      </div>
    </div>
  </div>
  <div
    class="za-mask normal"
  />
</div>
`;
